src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetAnnotationExecutor.java

```diff
@@ -49,40 +54,63 @@
-			
-			List<Long> datasetIdsLoaded = new ArrayList<>();
+
+			List<Long> htpAnnotationsIdsLoaded = new ArrayList<>();
+			List<Long> previousIds = htpExpressionDatasetAnnotationService.getAnnotationIdsByDataProvider(dataProvider.name());
 			
 			bulkLoadFileHistory.setTotalRecords((long) htpExpressionDatasetData.getData().size());
+			boolean success = runLoad(bulkLoadFileHistory, dataProvider, htpExpressionDatasetData.getData(), htpAnnotationsIdsLoaded);
+			if(success) {
+				runCleanup(htpExpressionDatasetAnnotationService, bulkLoadFileHistory, dataProvider.name(), previousIds, htpAnnotationsIdsLoaded, fms.getFmsDataType());
+			}
+			bulkLoadFileHistory.finishLoad();
 			updateHistory(bulkLoadFileHistory);
-
-			boolean result = runLoaddatasetid(externalDataBaseEntityService, bulkLoadFileHistory, dataProvider, htpExpressionDatasetData.getData(), datasetIdsLoaded, false);
 		} catch (Exception e) {
 			failLoad(bulkLoadFileHistory, e);
 			e.printStackTrace();
 		}
 	}
 
-	private boolean runLoaddatasetid(ExternalDataBaseEntityService externalDataBaseEntityService, BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDatasetData, List<Long> datasetIdsLoaded, boolean isUpdate) {
+	private boolean runLoad(BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDataset, List<Long> htpAnnotationsIdsLoaded) {
 		ProcessDisplayHelper ph = new ProcessDisplayHelper();
 		ph.addDisplayHandler(loadProcessDisplayService);
-		ph.startProcess("External Database Entity DTO Update for " + dataProvider.name(), htpDatasetData.size());
-		for (HTPExpressionDatasetAnnotationFmsDTO dto : htpDatasetData) {
+		ph.startProcess("HTP Expression Dataset Annotation DTO Update for " + dataProvider.name(), htpDataset.size() * 2);
+		boolean isSuccess = true;
+		for (HTPExpressionDatasetAnnotationFmsDTO dto : htpDataset) {
 			try {
-				ExternalDataBaseEntity dbObject = externalDataBaseEntityService.upsert(dto.getDatasetId(), dataProvider);
+				HTPExpressionDatasetAnnotation dbObject = htpExpressionDatasetAnnotationService.upsert(dto, dataProvider);
 				history.incrementCompleted();
-				if (datasetIdsLoaded != null) {
-					datasetIdsLoaded.add(dbObject.getId());
+				if (dbObject != null) {
+					htpAnnotationsIdsLoaded.add(dbObject.getId());
 				}
 			} catch (ObjectUpdateException e) {
 				history.incrementFailed();
 				addException(history, e.getData());
 			} catch (Exception e) {
+				e.printStackTrace();
 				history.incrementFailed();
 				addException(history, new ObjectUpdateExceptionData(dto, e.getMessage(), e.getStackTrace()));
 			}
+			if (history.getErrorRate() > 0.25) {
+				Log.error("Failure Rate > 25% aborting load");
+				finalSaveHistory(history);
+				failLoadAboveErrorRateCutoff(history);
+			}
+			ph.progressProcess();
 		}
-		updateHistory(history);
+		updateHistory(history);		
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line must not have trailing whitespace
<!-- __reviewdog__:ChA2YmFkNWQxYmE2ZGZhYmZlEgpjaGVja3N0eWxl -->